cumulusci/tasks/release_notes/task.py

```diff
@@ -55,3 +73,127 @@ def _run_task(self):
+        branch_name = self.options.get("branch_name")
+        parent_branch_name = self.options.get("parent_branch_name")
+
+        if (not branch_name and not parent_branch_name) or (
+            branch_name and parent_branch_name
+        ):
+            raise TaskOptionsError(
+                "You must specify either branch_name or (exclusive) parent_branch_name."
+            )
+
+        self.repo = self.get_repo()
+        generator = ParentPullRequestNotesGenerator(
+            self.github, self.repo, self.project_config
+        )
+
+        if branch_name:
+            self._handle_branch_name_option(generator, branch_name)
+        elif parent_branch_name:
+            self._handle_parent_branch_name_option(generator, parent_branch_name)
+
+    def _handle_branch_name_option(self, generator, branch_name):
+        parent_pull_request = None
+        pull_request = get_pull_request_by_branch_name(self.repo, branch_name)
+
+        if not pull_request:
+            self.logger.info(
+                "Pull request not found for branch: {}.\nExiting...".format(branch_name)
+            )
+            return
+
+        base_branch = pull_request.base.ref
+        feature_prefix = self.project_config.project__git__prefix_feature
+        if not base_branch.startswith(feature_prefix):
+            self.logger.info("Pull request's base is not a feature branch.\nExiting...")
+            return
+
+        parent_pull_request = self._get_parent_pull_request(base_branch)
+        if is_label_on_pull_request(
+            self.repo, parent_pull_request, self.BUILD_NOTES_LABEL
+        ):
+            generator.aggregate_child_change_notes(parent_pull_request)
+        else:
+            generator.update_unaggregated_pr_header(parent_pull_request, branch_name)
+
+    def _handle_parent_branch_name_option(self, generator, parent_branch_name):
+        pull_request = get_pull_request_by_branch_name(self.repo, parent_branch_name)
+        if not pull_request:
+            # If we don't find a pull request for a specified parent
+            # branch we don't create one. Notify the user, and exit
+            self.logger.info(
+                "No pull request found for branch: {}.\nExiting...".format(
+                    parent_branch_name
+                )
+            )
+        elif is_label_on_pull_request(self.repo, pull_request, self.BUILD_NOTES_LABEL):
+            # We can only aggregate child change notes when given the parent_branch option
+            #
+            # We aren't able to append to the 'Unaggregated Pull Reqeusts' header.
+            # We don't know at what time the label was applied to the pull request;
+            # and therefore, we cannot determine which child pull requests are already
+            # aggregated into the parent pull request, and which ones should be
+            # included in the 'Unaggregated Pull Requests' section.
+            generator.aggregate_child_change_notes(pull_request)
+        else:
```

Comment:
Since running the task with parent_branch_name specified is intended as a manual task only, I'm sort of inclined to say we should go ahead and aggregate the child change notes whether or not the label is there. That way if a project doesn't want to use the label and automatic updates at all but still wants to aggregate manually from time to time, they can do so.

Reply:
I might be missing something here, but wouldn't that mean the following scenario would always be possible:
(1) Doc writer begins to make manual changes to the notes on a long running feature branch.
(2) After those manual changes occur a new commit comes through on a child branch
(3) The commit triggers this task to execute with the 'branch_name' option which no longer checks for the presence of the label, and automatically re-aggregates all of the change notes, which overwrites all of the manual changes that the writer made.

Also, there wouldn't be the need for the 'Unaggregated Pull Requests' section of the change notes.

Reply:
This comment is on the _handle_parent_branch_name_option method -- I was only proposing a change for when parent_branch_name is specified, not branch_name.